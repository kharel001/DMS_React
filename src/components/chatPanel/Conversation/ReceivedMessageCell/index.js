import React from 'react';

const ReceivedMessageCell = ({conversation, user}) => {
  return (
    <div className="d-flex flex-nowrap chat-item">

      <img className="rounded-circle avatar size-40 align-self-end" src={user.thumb}
           data-test="imageComp" alt=""/>

      <div className="bubble">
        <div className="message" data-test="messageComp">{conversation.message}</div>
        <div className="time text-muted text-right mt-2" data-test="sentAtComp">{conversation.sentAt}</div>
      </div>

    </div>
  )
};

export default ReceivedMessageCell;