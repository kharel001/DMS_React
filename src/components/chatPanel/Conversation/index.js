import React from 'react';
import ReceivedMessageCell from "./ReceivedMessageCell/index";
import SentMessageCell from "./SentMessageCell/index";

const Conversation = ({conversationData, selectedUser}) => {

  return (
    <div className="chat-main-content">
      {conversationData.map((conversation, index) => conversation.type === 'sent' ?
        <SentMessageCell key={index} conversation={conversation} data-test="sentMessageComp"/> :
        <ReceivedMessageCell key={index} conversation={conversation} user={selectedUser} data-test="receivedMailComp"/>
      )}
    </div>
  )
};

export default Conversation;