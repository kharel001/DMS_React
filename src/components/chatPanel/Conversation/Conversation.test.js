import React from 'react';
import Conversation from './index';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const dummy_data = {
    conversationData:[{
        id:1, data:"hello1", type:"sent"
    },
    {
        id:2, data:"hello2", type:"received"
    },
    {
        id:3, data:"hello3",type:"sent"
    }],
    selectedUser:"user1"
}
const setup = (props = {}) => {
    return shallow( < Conversation
     {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

{/* it("Custom Scrollable render", () => {
        let wrapper = setup(dummy_data);
        // expect(wrapper).toMatchSnapshot();
}) */}
it("Snapshot Conversation", () => {
    let wrapper = setup(dummy_data);
    expect(wrapper).toMatchSnapshot();
    // const span2= wrapper2.find('div')
    // expect(span2).toHaveLength(5); //doesn't work as expected
    // expect(img1.props("src")).toEqual("https://via.placeholder.com/150x150")
    
})



it("Components Conversation", () => {
    let wrapper = setup(dummy_data);
    let sentMessageComp = findByTestAttr(wrapper, "sentMessageComp");
    let receivedMailComp = findByTestAttr(wrapper, "receivedMailComp");
    expect(receivedMailComp).toHaveLength(1);
    expect(sentMessageComp).toHaveLength(2);
    expect(sentMessageComp.at(0).prop('conversation')).toEqual({
        id:1, data:"hello1",type:"sent"
    });

    expect(sentMessageComp.at(1).prop('conversation')).toEqual(
    {
        id:3, data:"hello3",type:"sent"
    });

    expect(receivedMailComp.at(0).prop('conversation')).toEqual(
    {
        id:2, data:"hello2",type:"received"
    });


    expect(receivedMailComp.at(0).prop('user')).toEqual("user1")


})