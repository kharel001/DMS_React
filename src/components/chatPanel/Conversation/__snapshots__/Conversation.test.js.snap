// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Conversation 1`] = `ShallowWrapper {}`;
